common/lcs/src/ser.rs

```diff
@@ -79,13 +79,28 @@ impl Serializer {
         self.output
     }
 
+    fn serialize_u32_as_uleb128(&mut self, mut value: u32) -> Result<()> {
+        use serde::ser::Serializer;
+        while value >= 0x80 {
+            // Write 7 (lowest) bits of data and set the 8th bit to 1.
+            let byte = (value & 0x7f) as u8;
+            self.serialize_u8(byte | 0x80)?;
+            value >>= 7;
+        }
+        // Write the remaining bits of data and set the highest bit to 0.
+        self.serialize_u8(value as u8)
+    }
+
+    fn serialize_variant_index(&mut self, v: u32) -> Result<()> {
```

Comment:
varint not variant?

Reply:
no, I meant variant index (the tag of an enum in Serde's speech)